### ISSUE TITLE
small screen capability

### ISSUE THREAD

**Issue by pljakobs** [Shopify/dashing#608]
I'm trying to use dashing on a 2.8" RaspberryPi screen as an IoT display (running chromium kiosk mode).

I've looked around but wasn't able to find a way to resize the whole experiece (I would like to see 2x3 widgets, that is, every cell would be 120x120px - padding)

**Comment by tylermauthe** [comment]
Dashing isn't designed very responsively, as the project was created to be run on TVs of a specific size.

I guess you could throw a CSS zoom on the body to resize it.

I think making Dashing resize more automagically via Responsive Design would be a cool feature though. Would that be something you'd be interested in?

@pushmatrix -- What do you think about responsifying Dashing?

**Comment by mapledyne** [comment]
We put something like this at the top of our erb template on dashboards we want different sizes. We don't use these on mobile, but the concept should be the same.

In addition, we found that some widgets (maxmeter for instance) have minimum sizes they expect. We had to derive from them and make smaller version of the widgets so they looked good in the smaller boxes.

    <script type='text/javascript'>
    $(function() {
      // These settings override the defaults set in application.coffee. You can do this on a per dashboard basis.
      
      Dashing.widget_base_dimensions = [200, 250]
      Dashing.widget_margins = [5, 5]
      Dashing.numColumns = 6
      });
    </script>

**Comment by pushmatrix** [comment]
The main use cases I've seen for Dashing have been to design for a single screen in mind (ex: a tv). If you want to optimize for smaller screens, you can set the dimensions using something like @mapledyne mentioned. There may be some css needing to be tweaked.

Dashing was created so that you pick the exact spots where widgets will be. After all, it's nice to always see the widgets in the place you expect them. If it becomes responsive, then suddenly you lose some of that. I'm open for convincing though.

**Comment by mapledyne** [comment]
I agree with @pushmatrix on the scope of Dashing - it handles one-screen layouts quite well. Making it responsive definitely is a bit of a stray away from its core competency. There are lots of things I'd love to see improved in Dashing (but don't get me wrong - it's a fantastic tool), but making it responsive wouldn't help us in the slightest, and we have thousands of widgets and hundreds of dashboards.
